app/operations/workflows/unretire_subjects.rb

```diff
@@ -17,7 +21,14 @@ def execute
     end
 
     def cached_subject_ids
-      @cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids ||= Array.wrap(@subject_ids) | Array.wrap(@subject_id)
+      cached_subject_ids.push(*subject_set_subject_ids) unless subject_set_subject_ids.empty?
+      @cached_subject_ids = cached_subject_ids
+    end
+
+    def subject_set_subject_ids
+      set_ids ||= Array.wrap(@subject_set_ids) | Array.wrap(@subject_set_id)
```

Comment:
no need to memoize here
```suggestion
      set_ids = Array.wrap(@subject_set_ids) | Array.wrap(@subject_set_id)
```